src/fontra/core/packedpath.py

```diff
@@ -240,74 +241,20 @@ def endPath(self):
         self._currentContour = None
 
     def addComponent(self, glyphName, transformation, **kwargs):
-        from .classes import Component, Transformation
-
-        xx, xy, yx, yy, dx, dy = transformation
-        rotation, scaleX, scaleY, skewX, skewY = decomposeTwoByTwo((xx, xy, yx, yy))
-        # TODO rotation is problematic with interpolation: should interpolation
-        # go clockwise or counter-clockwise? That ambiguous, and get more complicated
-        # with > 2 masters. Perhaps we can "normalize" the rotations angles in some
-        # way to have reasonable behavior in common cases.
-        if rotation == -0.0:
-            rotation = 0.0
-
-        transformation = Transformation(
-            translateX=dx,
-            translateY=dy,
-            rotation=math.degrees(rotation),
-            scaleX=scaleX,
-            scaleY=scaleY,
-            skewX=math.degrees(-skewX),
-            skewY=math.degrees(skewY),
-            tCenterX=0,
-            tCenterY=0,
-        )
+        from .classes import Component
 
+        transformation = DecomposedTransform.fromTransform(transformation)
         self.components.append(Component(glyphName, transformation))
 
     def addVarComponent(
         self, glyphName, transformation, location, identifier=None, **kwargs
     ):
-        from .classes import Component, Transformation
+        from .classes import Component
 
-        # TODO: https://github.com/googlefonts/fontra/issues/245
-        transformation = Transformation(**asdict(transformation))
+        transformation = DecomposedTransform(**asdict(transformation))
```

Comment:
Ah, here we were receiving a `DecomposedTransform` and we needed it to be a `Transformation`. So strictly this isn't necessary anymore. It may still be good to make a copy of the object, though, but let's use `copy()` (from the `copy` module) for that.

Reply:
I sent a patch but I noticed that unit tests do not cover this function currently. I am not sure if it works correctly.

Reply:
Ah yes, I see. Let me think. I can test it manually for now, but indeed we'd need a test case. It would need a special constructed font, though, which we currently don't have in the repo.